src/anemoi/graphs/nodes/attributes.py

```diff
@@ -148,3 +151,105 @@ def get_raw_values(self, nodes: NodeStorage, *args, **kwargs) -> np.ndarray:
             np.array(area_weights).sum(),
         )
         return area_weights
+
+
+class BooleanBaseNodeAttribute(BaseNodeAttribute, ABC):
+    """Base class for boolean node attributes."""
+
+    def __init__(self) -> None:
+        super().__init__(norm=None, dtype="bool")
+
+
+class MissingZarrVariable(BooleanBaseNodeAttribute):
+    """Mask of missing values of a Zarr dataset variable.
+
+    It reads a variable from a Zarr dataset and returns a boolean mask of missing values in the first timestep.
+
+    Attributes
+    ----------
+    variable : str
+        Variable to read from the Zarr dataset.
+    norm : str
+        Normalization of the weights.
+
+    Methods
+    -------
+    compute(self, graph, nodes_name)
+        Compute the attribute for each node.
+    """
+
+    def __init__(self, variable: str) -> None:
+        super().__init__()
+        self.variable = variable
+
+    def get_raw_values(self, nodes: NodeStorage, **kwargs) -> np.ndarray:
+        assert (
+            nodes["node_type"] == "ZarrDatasetNodes"
+        ), f"{self.__class__.__name__} can only be used with ZarrDatasetNodes."
+        ds = open_dataset(nodes["_dataset"], select=self.variable)[0].squeeze()
+        return np.isnan(ds)
+
+
+class NotMissingZarrVariable(MissingZarrVariable):
```

Comment:
Is that not the inverse of the MissingZarrVariable?

Reply:
Yes!